src/utils/pgwire/src/pg_extended.rs

```diff
@@ -0,0 +1,108 @@
+use bytes::{Bytes, BytesMut};
+use regex::Regex;
+
+use crate::pg_field_descriptor::{PgFieldDescriptor, TypeOid};
+
+// NOTE : Code dundancy , may need to modify later
+fn cstr_to_str(b: &Bytes) -> &str {
+    let without_null = if b.last() == Some(&0) {
+        &b[..b.len() - 1]
+    } else {
+        &b[..]
+    };
+    std::str::from_utf8(without_null).unwrap()
+}
+
+pub struct pg_statement {
+    name: String,
+    query_string: Bytes,
+    type_description: Vec<TypeOid>,
+    row_description: Vec<PgFieldDescriptor>,
+}
+
+impl pg_statement {
+    pub fn new(
+        name: String,
+        query_string: Bytes,
+        type_description: Vec<TypeOid>,
+        row_description: Vec<PgFieldDescriptor>,
+    ) -> Self {
+        // LIMIT: the number of type_description should equal to the number of generic parameter
+        {
+            let parameter_parttern = Regex::new(r"\$[0-9][0-9]*").unwrap();
+            let s = cstr_to_str(&query_string);
+            let count = parameter_parttern.find_iter(s).count();
+            assert_eq!(count, type_description.len());
+        }
+        pg_statement {
+            name,
+            query_string,
+            type_description,
+            row_description,
+        }
+    }
+
+    pub fn get_name(&self) -> String {
+        self.name.clone()
+    }
+
+    pub fn get_type_desc(&self) -> Vec<TypeOid> {
+        self.type_description.clone()
+    }
+
+    pub fn get_row_desc(&self) -> Vec<PgFieldDescriptor> {
```

Comment:
```suggestion
    pub fn row_desc(&self) -> Vec<PgFieldDescriptor> {
```

Reply:
should I change all 'get_*()' method into  '*()'? or just change what you point out.

Reply:
You can do this to the code affected in this PR. For other code, we can submit a refactor PR